packages/lit-dev-content/site/tutorials/content/tooltip/08.md

```diff
@@ -0,0 +1,66 @@
+The tooltip supports good performance via lazy creation, but using the
+`lazy` API is cumbersome. In addition, as implemented, the tooltip content
+doesn't update when the greeted person's name changes.
+
+A Lit directive can address these issues by encapsulating the tooltip's `lazy`
+API. Users of the directive will use an
+[element expression](http://lit.dev/docs/templates/expressions/#element-expressions)
+on the target element and provide a
+[TemplateResult](http://lit.dev/docs/libraries/standalone-templates/#rendering-lit-html-templates)
+containing the tooltip content.
+
+A scaffold for the directive is setup for you to fill in. The steps are:
+
+1. Make a `TemplateResult` that will render a tooltip with the given content.
+1. When the directive is first updated, add a call to the `lazy` API that
+renders the `TemplateResult` into a `ChildPart` called `renderPart` and returns
+the tooltip element.
+1. If the tooltip has already rendered (the `renderPart` exists), update the
+part value with the new `result`.
+
+```ts
+class TooltipDirective extends Directive {
+  isSetup = false;
+  renderPart?: ChildPart;
+  render(value: unknown = '') {}
+  update(part: ElementPart, [value]: Parameters<this['render']>) {
+    // 1. Make a TemplateResult that will render a tooltip with provided content.
+    const result = html`<simple-tooltip>${value}</simple-tooltip>`;
+    if (!this.isSetup) {
+      this.isSetup = true;
+      // 2. Call the `lazy` API and render and return a tooltip,
+      // saving the `renderPart`.
+      SimpleTooltip.lazy(part.element, () => {
+        const fragment = document.createDocumentFragment();
+        this.renderPart = render(result, fragment, part.options);
+        return fragment.firstElementChild as SimpleTooltip;
+      });
+    // 3. If the tooltip has rendered, update it by setting `renderPart`'s value.
+    } else if (this.renderPart) {
+      setChildPartValue(this.renderPart, result);
+    }
+  }
```

Comment:
So, my initial reaction to this is that this is a pretty unnatural factoring if you just came at this from a directive from the beginning, without the `lazy` thingy first, so it sort of obscures a more elegant way to write this as a directive.

It'd be nice if we could just call 
```ts
render(html`<simple-tooltip>${value}</simple-tooltip>`, part.element.nextSibling, {renderBefore: true})
```
but only one directive could do this since that hangs an expando for the part off the reference element. Sort of unsatisfying.

Reply:
Seems like it would be nice to support passing in the `part` in render.

`this.renderPart = render(value, target.parentNode, {part: this.renderPart, renderBefore: target.nextSibling});`

Reply:
Perhaps this is better?

```
const renderBefore = target.parentNode.insertBefore(document.createComment(), target.nextSibling);
render(value, target.parentNode, {renderBefore});
```

Reply:
Simplified to just have the user render the content into the tooltip, avoiding any of these issues.